_posts/2023-02-08-.md

```diff
@@ -0,0 +1,86 @@
+
+We named this new product FESK (Front-End Starter Kit). It contained everything an engineer needed to develop and deploy a web application, leveraging a pattern library and common utilities to solve customer problems faster than FEAR, using Vanilla JS.
+
+## So, what went wrong?
+
+On reflection we see this in three parts:
+
+### 1. Technology challenges
+
+We decided on Vanilla JS because of the instability of popular frameworks at the time; however, this eventually made it hard to onboard and recruit engineers as most were focused on learning the other popular and emerging frameworks (such as Angular, React, etc.) which were becoming more lucrative on the job market.
+
+FESK was complex to use and did not stick to industry standards, so as a direct result we had to maintain a lot of custom documentation.
+
+The framework was built as multiple modules, leveraging NPM packages. Unintentionally, but by design, it separated the module from the users’ application and left us unaware of what was being used and the impact of change if a component was updated. This resulted in an architecture that was not well thought out and each web app became a page template, with each page template requiring its own engineering team.
+
+With many separate teams as consumers, it was hard to understand if they were consuming the latest releases, and some teams didn’t update past the first release. Resulting in us missing the benefits of code reuse, and had many different independent forks to maintain.
+
+### 2. Company culture and buy in (support)
+
+FESK began as a POC for improving the front-end customer experience delivery, with two main focuses; Page speed and SEO value. Once initially proved, it was moved quickly to production and throttled up to the majority of the customer base without the usual due diligence or preparation for production.
+
+While developing FESK itself, it was difficult to understand the consumer problems as they couldn’t articulate how FESK might be able to help them. This resulted in production issues and a lack of confidence in the product and its rollout across the web channel. As more teams adopted the product, it became clear we had not taken the business, and more importantly, the engineers on the journey to develop, own and maintain it.
+
+This lack of buy-in showed itself through teams frequently not contributing back to FESK because of slowness in PR approvals, or simply not understanding its usefulness in their day-to-day. In addition, UX came late to the party as it was engineering led, so they were playing catch up to requests they maybe did not fully understand.
+
+As it grew, in an attempt to encourage engineers to talk about their common platform and progress it, the centralised team was replaced with a community model. This failed as all the users did not know what to do. Subsequently, without an active community, teams ended up building siloed components and experiences, reinventing the wheel without realising.
```

Comment:
`This failed as all the users did not know what to do.`.  What did they not know to do?